saleor/graphql/order/mutations/orders.py

```diff
@@ -671,3 +672,42 @@ def perform_mutation(cls, _root, info, **data):
         instance.status = InvoiceStatus.READY
         instance.save()
         return UpdateInvoice(invoice=instance)
+
+
+class SendInvoiceEmail(ModelMutation):
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of an invoice to be sent.")
+
+    class Meta:
+        description = "Send an invoice by email."
+        model = models.Invoice
+        permissions = (OrderPermissions.MANAGE_ORDERS,)
+        error_type_class = InvoiceError
+        error_type_field = "invoice_errors"
+
+    @classmethod
+    def clean_instance(cls, info, instance):
+        validation_errors = {}
+        if instance.status != InvoiceStatus.READY:
+            validation_errors["invoice"] = ValidationError(
+                "Invoice needs to be ready in order to send it.",
+                code=InvoiceErrorCode.INVALID,
+            )
+        for field in ("url", "number"):
+            if not getattr(instance, field):
+                validation_errors[field] = ValidationError(
```

Comment:
Similar as above, here we're matching the error with a particular field, but none of these fields were passed in the input. I would say this error means that the invoice object is in a state that doesn't allow us to perform this operation, so it's a general error. We also have `URL_OR_NUMBER_NOT_SET` error code from what I see (which I thought we dropped 🧐) - maybe it would make sense to use this code here.